changed-files/action.yml

```
@@ -25,9 +25,10 @@ runs:
       run: |
         BASE_BRANCH=${{ inputs.base-branch }}
         HEAD_BRANCH=${{ inputs.head-branch }}
+        HEAD_REF=${{ github.ref }}:$HEAD_BRANCH
 
         git fetch origin $BASE_BRANCH --depth 1
-        git fetch origin $HEAD_BRANCH --depth 1
+        git fetch origin $HEAD_REF --depth 1
```

```suggestion
        git fetch origin ${{ inputs.base_ref }}:refs/heads/changed_base --depth 1
        git fetch origin ${{ inputs.head_ref }}:refs/heads/changed_head --depth 1
```

What this does is tells `git` to fetch things from`origin` to local refs.

Below, you would then do `git diff --name-only refs/heads/compare_base refs/heads/compare_head`.

---

This is good, thanks!

---

This is good, thanks!